src/uploaders/blossom.ts

```diff
@@ -23,6 +27,44 @@ export interface BlossomResponseError {
   message: string
 }
 
+export interface EncryptionResult {
+  key: string
+  nonce: string
+  ciphertext: Uint8Array
+}
+
+export async function encryptFile({encryptionAlgorithm, file}: BlossomOptions): Promise<EncryptionResult> {
+  if (encryptionAlgorithm !== 'aes-gcm') {
+    throw new Error(`Unsupported encryption algorithm: ${encryptionAlgorithm}`)
+  }
+
```

Comment:
What do you think to include noble/hashes and ciphers here and save some bundle size? I think this could work

```ts
  const [{ randomBytes }, { gcm }, { bytesToHex }] = await Promise.all([
    import('@noble/ciphers/webcrypto'),
    import('@noble/ciphers/aes'),
    import('@noble/hashes/utils'),
  ])
```